wp-offline-shell/lib/service-worker.js

```diff
@@ -107,14 +87,14 @@
     // Manages returning responses from cache or the server
     onFetch: function(event) {
       var request = event.request;
-      var lookupRequest = this.normalizeAndAnonymize(request);
-      if (!this.shouldBeHandled(lookupRequest)) {
+      if (!this.shouldBeHandled(request)) {
         return;
       }
 
       event.respondWith(
-        caches.match(lookupRequest)
-          .then(response => {
+        caches.match(request, {
```

Comment:
What if you passed the cleansed URL instead of the request? So we don't have to use `ignoreSearch`.